src/bika/lims/browser/analysisrequest/add2.py

```diff
@@ -524,11 +524,21 @@ def get_services(self, poc="lab"):
         analyses[""] = []
 
         for brain in services:
-            category = brain.getCategoryTitle
+            category = self.get_category_title(brain)
             if category in analyses:
                 analyses[category].append(brain)
         return analyses
 
+    def get_category_title(self, service):
+        """Return the title of the category the service is assigned to
+        """
+        service = api.get_object(service)
+        cat_uid = service.getRawCategory()
+        cat = self.get_object_by_uid(cat_uid)
```

Comment:
Here as well. You could consider returning immediately or maybe do this to skip the `self.get_object_by_uid` call:
```python
cat = self.get_object_by_uid(cat_uid) if cat_uid else None
```